catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -845,15 +591,449 @@ function DropzoneMessage({ error, warn, disabled }: DropzoneMessageProps) {
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    flexGrow: 1,
+    marginTop: t.spacing(2),
+    overflowY: 'auto',
+    position: 'relative',
+  },
+  outlined: {
+    outline: `2px dashed ${t.palette.primary.main}`,
+    outlineOffset: '-2px',
+  },
+}))
+
+type ContentsContainerProps = {
+  outlined?: boolean
+} & React.HTMLAttributes<HTMLDivElement>
+
+function ContentsContainer({ outlined, className, ...props }: ContentsContainerProps) {
+  const classes = useContentsContainerStyles()
+  return (
+    <div
+      className={cx(className, classes.root, outlined && classes.outlined)}
+      {...props}
+    />
+  )
+}
+
+const useContentsStyles = M.makeStyles((t) => ({
+  root: {
+    border: `1px solid ${t.palette.action.disabled}`,
+    borderRadius: t.shape.borderRadius,
+    display: 'flex',
+    flexDirection: 'column',
+    flexGrow: 1,
+    minHeight: 80,
+    outline: 'none',
+    overflow: 'hidden',
+  },
+  interactive: {
+    cursor: 'pointer',
+  },
+  active: {
+    background: t.palette.action.selected,
+  },
+  err: {
+    borderColor: t.palette.error.main,
+  },
+  warn: {
+    borderColor: t.palette.warning.dark,
+  },
+}))
+
+type ContentsProps = {
+  interactive?: boolean
+  active?: boolean
+  error?: boolean
+  warn?: boolean
+} & React.HTMLAttributes<HTMLDivElement>
+
+function Contents({
+  interactive,
+  active,
+  error,
+  warn,
+  className,
+  ...props
+}: ContentsProps) {
+  const classes = useContentsStyles()
+  return (
+    <div
+      className={cx(
+        className,
+        classes.root,
+        interactive && classes.interactive,
+        active && classes.active,
+        error && classes.err,
+        !error && warn && classes.warn,
+      )}
+      {...props}
+    />
+  )
+}
+
+type FileUploadProps = tagged.ValueOf<typeof FilesEntry.File> & {
+  prefix?: string
+  dispatch: DispatchFilesAction
+}
+
+function FileUpload({ name, state, size, prefix, dispatch }: FileUploadProps) {
+  const path = (prefix || '') + name
+
+  const handle = React.useCallback(
+    (cons: tagged.ConstructorOf<typeof FilesAction>) => (e: React.MouseEvent) => {
+      e.stopPropagation()
+      dispatch(cons(path))
+    },
+    [dispatch, path],
+  )
+
+  const action = React.useMemo(
+    () =>
+      ({
+        added: { hint: 'Remove', icon: 'clear', handler: handle(FilesAction.Revert) },
+        modified: {
+          hint: 'Revert',
+          icon: 'undo',
+          handler: handle(FilesAction.Revert),
+        },
+        hashing: {
+          hint: 'Revert',
+          icon: 'undo',
+          handler: handle(FilesAction.Revert),
+        },
+        deleted: {
+          hint: 'Restore',
+          icon: 'undo',
+          handler: handle(FilesAction.Revert),
+        },
+        unchanged: { hint: 'Delete', icon: 'clear', handler: handle(FilesAction.Delete) },
+      }[state]),
+    [state, handle],
+  )
+
+  const onClick = React.useCallback((e: React.MouseEvent) => {
+    e.stopPropagation()
```

Comment:
It worth writing down the reason in comments

Reply:
`// stop click from propagating to parent elements and triggering their handlers`

Reply:
The next team member will need to add some components above in the hierarchy, and he will have bad times finding why his `onClick` handler doesn't work. Finally, found this `stopPropagation` he will need some glue to refactor this and not to break already existing code. You can help him by providing the comment, explaining the reason behind this architectural decision.

If you don't understand why it's important, don't bother, we can get back to this when the problem becomes real if it does.

Reply:
but i'm not arguing here -- i've added the comments)
or do you mean these comments should be more detailed?

Reply:
I'm asking for
```js
// use Ohm's law to calculate voltage,
// see ohm-i.tsx and ohm-r.tsx
const u = i* r
```
rather then
```js
// assign `u` to the `i` multiplied by `r`, so `u` will be the product of `i` and `r` multiplication
const u = i * r
```
---
Again, if you don't understand what I mean, we can get back to this when the real problem occurs